### PR TITLE
Feat/git deps

### PR DESCRIPTION
**Pull request by nikhedonia**

### CODE REVIEW

File: buckaroo/Bash.fs [hunk 1]
```diff
@@ -7,9 +7,9 @@ open System.Threading.Tasks
 
 type ProgressCallback = string -> Unit
 
-let escapeBash (command : string) = 
-  if command.Contains("\"") || command.Contains("$")
```
Comment (njlr) [inline buckaroo/Bash.fs]: Keep up the great work! :+1:

File: buckaroo/Bash.fs [hunk 2]
```diff
@@ -54,16 +54,16 @@ let runBashSync (command : String) (stdoutHandler : ProgressCallback) (stderrHan
...
     Task.Factory.StartNew(startProcess)
     |> Async.AwaitTask
     |> Async.StartChild
 
   do! exitSignal
 
-  if p.ExitCode > 0
-  then 
-    return 
+  if p.ExitCode > 0 && p.ExitCode < 128
```
Comment (njlr) [inline buckaroo/Bash.fs]: I think we should leave it up to the caller if they want to throw or not. Does 128 always indicate success? I think this depends on the command you have run.

File: buckaroo/Lock.fs [hunk 3]
```diff
@@ -1,24 +1,24 @@
 namespace Buckaroo
 
 type LockedPackage = {
-  Version : Version; 
-  Location : PackageLocation; 
-  PrivatePackages : Map<PackageIdentifier, LockedPackage>; 
+  Version : Version;
```
Comment (njlr) [inline buckaroo/Lock.fs]: Good catch! :fishing_pole_and_fish:

File: buckaroo/PackageIdentifier.fs [hunk 4]
```diff
@@ -36,7 +35,8 @@ module PackageIdentifier =
 
 
   let gitHubPackageIdentifierParser = parse {
-    do! CharParsers.skipString "github.com/" <|> CharParsers.skipString "github+"
+    do! CharParsers.skipString "github.com/" 
+      <|> CharParsers.skipString "gh+"
```
Comment (njlr) [inline buckaroo/PackageIdentifier.fs]: This is unrelated

Comment (njlr) [reply buckaroo/PackageIdentifier.fs]: ... and breaking.

File: buckaroo/PackageSource.fs [hunk 5]
```diff
@@ -1,17 +1,36 @@
...
-  Type : ArchiveType option; 
+  Url : string;
+  StripPrefix : string option;
+  Type : ArchiveType option
 }
 
-type PackageSource = 
-| Http of HttpPackageSource
+type GitPackageSource = {
+  Uri : string
```
Comment (njlr) [inline buckaroo/PackageSource.fs]: Rename to `Git` or `Url` for consistency?

File: buckaroo/Bash.fs (hunk 2 above)
Comment (nikhedonia) [reply buckaroo/Bash.fs]: this is a workaround for a bug which will be addressed in a seperate PR
